docs/templates.md

```diff
@@ -0,0 +1,175 @@
+`token`                 | A randomly generated token 
+`web_client_location`   | Default web client to invite user to join via
+
+### Substitutions from homeservers
+
+All values that get sent to the `/_matrix/identity/v2/store-invite` API 
+endpoint are accessible as variables in invite templates.
+
+Invites sent from Synapse make the following available:
+
+Variable                | Contents 
+-----------             | -------- 
+`address`               | The email address being invited (same as `to`)
+`bracketed_room_name`   | The string stored in `room_name` with brackets around it
+`bracketed_verified_sender`      | The string stored in `sender` with brackets around it
+`medium`                | The string "email"
+`org.matrix.msc3288.room_type`   | **UNSTABLE.** The same as `room_type`
+`org.matrix.web_client_location` | **UNSTABLE.** The same as `web_client_location`
+`room_alias`            | An alias for the room (probably more readable than `room_id`)
+`room_avatar_url`       | The URL of the room's avatar
+`room_id`               | The ID of the room to which the user is invited
+`room_join_rules`       | The join rules of the email (e.g. "public" or "invite")
+`room_name`             | The m.room.name state of the room. (e.g. "Synape Admins")
+`room_type`             | Is set to "m.space" if the invite is to a space
+`sender`                | The user ID of the inviter
+`sender_avatar_url`     | The URL of the inviter's avatar
+`sender_display_name`   | The current display name of the inviter
+
+Version when table last updated: Synapse v1.42.0
+
+
+## Verification email templates
+
+Verification template files should have the name `verification_template.eml.j2`.
+
+### All substitutions
+
+Variable                | Contents 
+-----------             | -------- 
+`date`                  | The time and date of sending as defined in RFC 2822 (e.g. "Fri, 09 Nov 2001 01:08:47 -0000") 
+`from`                  | The sending email address as configured in `email.from`
+`ipaddress`             | The IP address that the verification request came from. If the IP address is unknown then this is the string "an unknown location"
+`messageid`             | The unique ID for this email 
+`multipart_boundary`    | Randomized multipart boundary to use in multipart emails
+`to`                    | The destination email address 
+`token`                 | A randomly generated token that some clients might need
+
+
+## Verification SMS templates
+
+The SMS template isn't read from a file. Instead the SMS template
+should be placed as a string into the config option `bodyTemplate`.
+
+This string uses python `{variable}` substitution.
+
+### All substitutions
+
+Variable    | Contents 
+----------- | -------- 
+`token`     | A randomly generated token for the user to enter into the client
+
+
+## Migration email templates
+
+Migration template files should have the name `migration_template.eml.j2`.
```

Comment:
It might be worth double mentioning that this must use jinja, there's no `.eml` version.